src/doc/sprite.h

```diff
@@ -258,6 +260,9 @@ namespace doc {
     mutable RgbMapAlgorithm m_rgbMapAlgorithm;
     mutable std::unique_ptr<RgbMap> m_rgbMap;
 
+    // Last 'color fit criteria' used on indexed conversions
+    mutable FitCriteria m_fitCriteria = FitCriteria::DEFAULT;
```

Comment:
Do we really need this in `doc::Sprite`?

Reply:
You're right! We don't need it, and we don't need `mutable RgbMapAlgorithm m_rgbMapAlgorithm;` either. I'll add this parameters into class Rgbmap.

Reply:
Actually moving it to the RgbMap is a similar case. If you think it makes more sense to move them inside RgbMap go for it (not sure if this FitCriteria is only used inside the RgbMap and not in the findBest of the palette).

One important thing I'm missing and probably I thought incorrectly was that this FitCriteria was something used in the *Sprite > Color Mode > More Options > Advance* only. But it looks like you are planning as part of the whole program configuration. Something particular for the Octree (if I'm guided by the #2787 title), or something that can be used in both rgbmaps (rgb5a3/octree). What is the plan with this PR?

Reply:
In a first approximation, I thought of it only for octree quantization. But I tested it with RGB5A3 and it can be processed without problems. So I extended it for both rgbmaps.
I added a global setting to set the default value in Advanced conversion `RGB->Indexed`, then added this setting to work with `Sprite > ColorMode > Indexed` too.

We can stop here and wait for the artist's reception to see if it is feasible to extend this option further (include parameter in lua for example).

All of these other 'color fit criteria' in the `Palette::findBestFit` function can be useful for artists who want to fit any RGB image to a custom palette and achieve a more appropriate color selection.

Reply:
@dacap I will limit the feature to be used only in "Sprite > Color Mode > Advanced"